tests/test_verify_build.py

```diff
@@ -895,3 +895,127 @@ def test_install_with_target_ignore_installed(pipexec, pyexec, build):
     out = proc.stdout.decode()
     assert "installed cffi" in out
     assert "already satisfied: cffi" not in out
+
+
+def test_no_legacy_hashlib(pipexec, pyexec, build):
+    """
+    Verify hashlib can find the legacy openssl provider.
+    """
+    env = {"OPENSSL_CONF": str(build / "openssl.cnf")}
+    with open(env["OPENSSL_CONF"], "w") as fp:
+        fp.write(
+            textwrap.dedent(
+                """
+            HOME			= .
+            openssl_conf = openssl_init
+            [openssl_init]
+            providers = provider_sect
+            [provider_sect]
+            default = default_sect
+            [default_sect]
+            activate = 1
+            """
+            )
+        )
+    proc = subprocess.run(
+        [
+            pyexec,
+            "-c",
+            "import hashlib; print(hashlib.algorithms_available)",
+        ],
+        check=True,
+        stdout=subprocess.PIPE,
+        env=env,
+    )
+    assert b"md4" not in proc.stdout
```

Comment:
What would be on the system OpenSSL which is not on our OpenSSL?
Can we assert something like that too?

Reply:
The two tests are confirming that md4 is missing when we have no legacy provider enabled and available when the provider is enabled. Since we're using the openssl.cnf from the tests there is no system openssl involved.